File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/C#/GUI/ButtonHoverSetText.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[RequireComponent(typeof(GUITexture))]

public class ButtonHoverSetText : MonoBehaviour {
	public GUIText chooseText;
	public string textValue;
	public float offset = 30;
	private Vector2 pixelOffset;

	void Update() {
		if(GetComponent<GUITexture>().HitTest(Input.mousePosition)) {
			chooseText.text = textValue;
			pixelOffset = chooseText.pixelOffset;
			pixelOffset.x = Input.mousePosition.x + offset;
			pixelOffset.y = Input.mousePosition.y;
			chooseText.pixelOffset = pixelOffset;
		}
	}
}

File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/JavaScript/GUI/ButtonHoverSetText.js
#pragma strict

@script RequireComponent(GUITexture);

public var chooseText : GUIText;
public var textValue : String;
public var fontType : Font;
public var offset : float = 30;

function Update() {
	if(GetComponent.<GUITexture>().HitTest(Input.mousePosition)) {
		chooseText.text = textValue;
		chooseText.font = fontType;
		chooseText.pixelOffset.x = Input.mousePosition.x + offset;
		chooseText.pixelOffset.y = Input.mousePosition.y;
	}
}